### PR TITLE
[Wallet]: Add hooks to query data from deepbook

### PR DESCRIPTION
**Pull request by plam-ml**
## Description 

https://mysten.atlassian.net/browse/APPS-1775

image

## Test Plan 

How did you test the new or updated feature?

---
If your changes are not user-facing and not a breaking change, you can skip the following section. Otherwise, please indicate what changed, and then add to the Release Notes section as highlighted during the release process.

### Type of Change (Check all that apply)

- [ ] protocol change
- [ ] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### CODE REVIEW

File: apps/wallet/src/ui/app/hooks/useDeepbook.ts [hunk 1]
```diff
@@ -0,0 +1,319 @@
...
+	side: 'ask' | 'bid' = 'ask',
+) {
+	return useQuery({
+		queryKey: [DEEPBOOK_KEY, 'get-prices', poolName, side],
+		queryFn: async () => {
+			const deepbookClient = await getDeepbookClient();
+			return deepbookClient.getLevel2BookStatus(
+				mainnetPools[poolName],
+				BigInt(0),
+				// TODO: need to switch back to 10n * DEFAULT_TICK_SIZE
```
Comment (plam-ml) [inline apps/wallet/src/ui/app/hooks/useDeepbook.ts:73]: This is not usable currently due to needing pagination from deepbook size. The query will fail with that upper threshold call

File: apps/wallet/src/ui/app/hooks/useDeepbook.ts [hunk 2]
```diff
@@ -0,0 +1,319 @@
...
+
+const FLOAT_SCALING_FACTOR = 1_000_000_000n;
+export const DEFAULT_TICK_SIZE = 1n * FLOAT_SCALING_FACTOR;
+const DEEPBOOK_KEY = 'deepbook';
+export const SUI_DIVISOR = 1_000_000;
+export const USDT_DIVISOR = 1000 * SUI_DIVISOR;
+export const OTHER_DIVISOR = 10 * SUI_DIVISOR;
+
+export const mainnetPools = {
+	SUI_USDC_1: '0x18d871e3c3da99046dfc0d3de612c5d88859bc03b8f0568bd127d0e70dbc58be',
```
Comment (Jibz-Mysten) [inline apps/wallet/src/ui/app/hooks/useDeepbook.ts:17]: what happens on testnet

**Review by Jibz-Mysten — APPROVED** [review]

File: apps/wallet/src/ui/app/hooks/useDeepbook.ts [hunk 3]
```diff
@@ -0,0 +1,319 @@
...
+		divisor = USDT_DIVISOR;
+	}
+
+	return new BigNumber(totalPrice.toString())
+		.dividedBy(prices.length)
+		.dividedBy(divisor)
+		.toNumber();
+}
+
+function useAvgPrice(base: Coins, quote: Coins) {
```
Comment (Jordan-Mysten) [inline apps/wallet/src/ui/app/hooks/useDeepbook.ts:105]: This hook feels super weird, why not just fetch what you need based on the base and quote, why are you fetching everything and then aggregating the hook response like this?

File: apps/wallet/src/shared/deepbook-client.ts [hunk 4]
```diff
@@ -0,0 +1,10 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { getActiveNetworkSuiClient } from '_shared/sui-client';
+import { DeepBookClient } from '@mysten/deepbook';
+
+export async function getDeepbookClient(): Promise<DeepBookClient> {
+	const suiClient = await getActiveNetworkSuiClient();
```
Comment (pchrysochoidis) [inline apps/wallet/src/shared/deepbook-client.ts:8]: Probably it's a bit confusing, but this `getActiveNetworkSuiClient` should not be used from the UI context. For the ui we have the `useSuiClient` to access the sui client.

This `getActiveNetworkSuiClient` is used in the background service

**Pull request merged**